bash_completion

```diff
@@ -2739,17 +2609,32 @@ _comp_xfunc()
     "$xfunc_name" "${@:3}"
 }
 
-_comp_deprecate_func _xfunc _comp_xfunc
-
 # source compat completion directory definitions
-_comp__init_compat_dir=${BASH_COMPLETION_COMPAT_DIR:-/etc/bash_completion.d}
-if [[ -d $_comp__init_compat_dir && -r $_comp__init_compat_dir && -x $_comp__init_compat_dir ]]; then
+_comp__init_compat_dirs=()
+if [[ -n $BASH_COMPLETION_COMPAT_DIR ]]; then
```

Comment:
I think we once normalized the style to `[[ $var ]]` (2984bd1183bfbd98d5e1239001f0319513b3b171). And I think there can be also `-o nounset`.

```suggestion
if [[ ${BASH_COMPLETION_COMPAT_DIR-} ]]; then
```